web/client/components/ScaleBar/ScaleCombo.jsx

```diff
@@ -0,0 +1,187 @@
+var MenuItem = ReactBts.MenuItem;
+
+/**
+ * [ScaleComboController new React Controller for ScaleCombo]
+ */
+var ScaleComboController = React.createClass(
+ {   /**
+     * [propTypes types of props]
+     * @type {Object}
+     */
+    propTypes: {
+        scalebox: React.PropTypes.object
+    },
+    /**
+     * [getDefaultProps fetch default props]
+     * @return {[Object]} [return scalebox object]
+     */
+    getDefaultProps() {
+        return {
+            scalebox: {
+                getComboItems: function() {
+                    var items = [[18, '18'], [17, '17'], [16, '16'], [15, '15'], [14, '14'], [13, '13'], [12, '12'], [11, '11'], [10, '10'], [9, '9'], [8, '8'], [7, '7'], [6, '6'], [5, '5'], [4, '4'], [3, '3'], [2, '2'], [1, '1'], [0, '0']];
+                    return items;
+                }
+            }
+        };
+    },
+    /**
+     * [getInitialState get initial state]
+     * @return {[Object]} [return showCombo, showGlyph, activeVal and items]
+     */
+    getInitialState() {
+        return {
+            showCombo: {
+                visibility: document.cookie && document.cookie.match('hideCombo') ? 'hidden' : 'visible'
+            },
+            showGlyph: document.cookie && document.cookie.match('hideCombo') ? 'chevron-left' : 'chevron-right',
+            activeVal: 'Example',
+            items: function() {
+                return;
+            }
+        };
+    },
+    /**
+     * [componentDidMount after component is loaded]
+     * @return {[undefined]} [no return]
+     */
+    componentDidMount() {
+        var that = this;
+
+        // list of items in scalecombo
+        var comboitems = that.props.scalebox.getComboItems();
+
+        // Disable dragging when user's cursor enters the element
+        that.getDOMNode().addEventListener('mouseover', function() {
+            that.props.scalebox.map.scrollWheelZoom.disable();
+            that.props.scalebox.map.doubleClickZoom.disable();
+        });
+
+        // Re-enable dragging when user's cursor leaves the element
+        that.getDOMNode().addEventListener('mouseout', function() {
+            that.props.scalebox.map.scrollWheelZoom.enable();
+            that.props.scalebox.map.doubleClickZoom.enable();
+        });
+
+        // start function that creates items as list of html elements
+        that.setComboItems(comboitems);
+
+        // when zoom ends trigger function that changes the scalecombo title
+        if (that.props.scalebox.map) {
+            that.props.scalebox.map.on('zoomend', function() {
+                that.setComboTitle(comboitems);
+            });
+        }
+    },
+    /**
+     * [render render view]
+     * @return {[HTML element]} [html]
+     */
+    render() {
+        return (
+            <div>
+                <DropdownButton className="mapstore-scalebox-combo-main" style={this.state.showCombo} title={this.state.activeVal} dropup pullRight>
+                {this.state.items()}
```

Comment:
I would prefer a more declarative approach. State is not needed here. In general we prefer stateless components wherever possible.